### ISSUE TITLE
Unable to render with including `plt.show()`

### ISSUE THREAD

**Issue by Mutaz94** [quarto-dev/quarto-cli#502]
Hi All;

I tried to take Quarto for whirl using Python 3. I use macOS Monetery with Python 3.10.

 I rendered the published example
```
title: "matplotlib demo"
format:
  html:
    code-fold: true
jupyter: python3
---

For a demonstration of a line plot on a polar axis, see @fig-polar.

```{python}
#| label: fig-polar
#| fig-cap: "A line plot on a polar axis"

import numpy as np
import matplotlib.pyplot as plt

r = np.arange(0, 2, 0.01)
theta = 2 * np.pi * r
fig, ax = plt.subplots(
  subplot_kw = {'projection': 'polar'} 
)
ax.plot(theta, r)
ax.set_rticks([0.5, 1, 1.5, 2])
ax.grid(True)
plt.show()
```

 I keep getting this error. 

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
/var/folders/7p/cp6pcdns68zfy1p7bchrfs4h0000gn/T/ipykernel_52212/1634554178.py in <module>
     13 ax.set_rticks([0.5, 1, 1.5, 2])
     14 ax.grid(True)
---> 15 plt.show()

/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/matplotlib/pyplot.py in show(*args, **kwargs)
    366     """
    367     _warn_if_gui_out_of_main_thread()
--> 368     return _backend_mod.show(*args, **kwargs)
    369
    370

/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/itermplot/__init__.py in show()
    138     figmanager = Gcf.get_active()
    139     if figmanager is not None:
--> 140         figmanager.show()
    141     else:
    142         for manager in Gcf.get_all_fig_managers():

/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/itermplot/__init__.py in show(self)
    296
    297         if hasattr(data, "getbuffer"):
--> 298             imgcat(data.getbuffer(), fn)
    299         else:  # Python 2
    300             imgcat(data.getvalue(), fn)

/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/itermplot/__init__.py in imgcat(data, fn)
    122         sys.stdout.buffer.write(buf)
    123     else:
--> 124         sys.stdout.write(buf)
    125     sys.stdout.flush()
    126

/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/ipykernel/iostream.py in write(self, string)
    509
    510         if not isinstance(string, str):
--> 511             raise TypeError(
    512                 f"write() argument must be str, not {type(string)}"
    513             )

TypeError: write() argument must be str, not <class 'bytes'>
TypeError: write() argument must be str, not <class 'bytes'>
```

It works well with R and Julia. This error appears only when trying `plt.show()`. Thanks!!

**Comment by jjallaire** [comment]
Your configuration seems to have the itermplot package configured as the Matplotlib backend (this is a package used for writing plots to iTerm2). I can see that from the second to last step in the callstack which references this code: https://github.com/daleroberts/itermplot/blob/master/itermplot/__init__.py

It looks like this is in turn configured via the `MPLBACKEND` environment variable: https://github.com/daleroberts/itermplot#installation

I'll see if there is a sane value we can force the `MPLBACKEND` to so that this doesn't occur. In the meantime it would be interesting if you confirmed that this is the root of the issue on your system.

**Comment by jjallaire** [comment]
It looks like this could also be set in your `matplotlibrc` config file: https://matplotlib.org/stable/users/explain/backends.html#selecting-a-backend

**Comment by jjallaire** [comment]
It looks like ipykernel sets the backend on startup of the Jupyter UI here: https://github.com/ipython/ipykernel/blob/main/ipykernel/kernelapp.py#L553-L554. It's not clear whether this code is executing in the Quarto case. Note however that if the env variable is already set it won't set it (so if the problem in your case is that the environment variable is set this wouldn't fix this issue).

**Comment by jjallaire** [comment]
Resolution on our end is to force the MPLBACKEND to use the Jupyter default backend by default: https://github.com/quarto-dev/quarto-cli/commit/980605f8278a623aaaf01d04473c7445cc557aa3

Note that other backends can still be used via `matplotlib.use()`.

This change is available in the v0.9.180 release of Quarto available here: https://github.com/quarto-dev/quarto-cli/releases/latest

If you add this code cell to your `.qmd` you should see what's going on with MPLBACKEND:

````markdown
```{python}
import os
print(os.getenv("MPLBACKEND"))
```
````

**Comment by Mutaz94** [comment]
Hi @jjallaire: Thanks for your detailed response. You are right, the issue is related to `MPLBACKEND` when I `unset MPLBACKEND` the issue will resolve.  itermplot was causing all this trouble. Thanks a lot!!
